ydb/library/binary_json/write.cpp

```diff
@@ -551,17 +555,81 @@ void DomToJsonIndex(const NUdf::TUnboxedValue& value, TBinaryJsonCallbacks& call
     }
 }
 
+void SimdJsonToJsonIndex(const simdjson::dom::element& value, TBinaryJsonCallbacks& callbacks) {
+    switch (value.type()) {
+        case simdjson::dom::element_type::STRING: {
+            std::string_view v;
+            Y_ABORT_UNLESS(value.get(v));
+            callbacks.OnString(v);
+            break;
+        }
+        case simdjson::dom::element_type::BOOL: {
+            bool v;
+            Y_ABORT_UNLESS(value.get(v));
+            callbacks.OnBoolean(v);
+            break;
+        }
+        case simdjson::dom::element_type::INT64: {
+            i64 v;
+            Y_ABORT_UNLESS(value.get(v));
+            callbacks.OnInteger(v);
+            break;
+        }
+        case simdjson::dom::element_type::UINT64: {
+            ui64 v;
+            Y_ABORT_UNLESS(value.get(v));
+            callbacks.OnUInteger(v);
+            break;
+        }
+        case simdjson::dom::element_type::DOUBLE: {
+            double v;
+            Y_ABORT_UNLESS(value.get(v));
+            callbacks.OnUInteger(v);
+            break;
+        }
+        case simdjson::dom::element_type::NULL_VALUE:
+            callbacks.OnNull();
+            break;
+        case simdjson::dom::element_type::ARRAY: {
+            callbacks.OnOpenArray();
+
+            simdjson::dom::array v;
+            Y_ABORT_UNLESS(value.get(v));
+            for (const auto& item : v) {
+                SimdJsonToJsonIndex(item, callbacks);
+            }
+
+            callbacks.OnCloseArray();
+            break;
+        }
+        case simdjson::dom::element_type::OBJECT: {
+            callbacks.OnOpenMap();
+
+            simdjson::dom::object v;
+            Y_ABORT_UNLESS(value.get(v));
+            for (const auto& item : v) {
+                callbacks.OnMapKey(item.key);
+                SimdJsonToJsonIndex(item.value, callbacks);
+            }
+
+            callbacks.OnCloseMap();
+            break;
+        }
+    }
+}
+
 }
 
 TMaybe<TBinaryJson> SerializeToBinaryJsonImpl(const TStringBuf json) {
-    TMemoryInput input(json.data(), json.size());
-    TBinaryJsonCallbacks callbacks(/* throwException */ false);
-    if (!ReadJson(&input, &callbacks)) {
+    simdjson::dom::parser parser;
```

Comment:
btw, ondemand should be faster than dom

Reply:
on benchmarks of SerializeToBinaryJson, both performed very closely, so I'm keeping dom for simpler API: https://gist.github.com/swalrus1/b3db37c1574756ff037349a91d9b155a